primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js

```diff
@@ -179,30 +179,68 @@ PrimeFaces.widget.BlockUI = PrimeFaces.widget.BaseWidget.extend({
+        // there can be 1 to N targets
+        for (var i = 0; i < this.target.length; i++) {
+            var currentTarget = $(this.target[i]),
+                currentTargetId = currentTarget.attr('id') || this.id,
+                currentContent = this.jq;
 
-        if (this.cfg.styleClass) {
-            this.blocker.addClass(this.cfg.styleClass);
-        }
+            // create a specific blocker for this target
+            var currentBlocker = $('<div id="' + currentTargetId + '_blocker" class="ui-blockui ui-widget-overlay ui-helper-hidden"></div>');
 
-        if (this.target.hasClass('ui-corner-all')) {
-            this.blocker.addClass('ui-corner-all');
-        }
+            // style the blocker
+            if (this.cfg.styleClass) {
+                currentBlocker.addClass(this.cfg.styleClass);
+            }
+            if (currentTarget.hasClass('ui-corner-all')) {
+                currentBlocker.addClass('ui-corner-all');
+            }
 
-        if (this.target.length > 1) {
-            this.content = this.content.clone();
-        }
+            // when more than 1 target need to clone the blocker for each target
+            if (isMultiple) {
+                currentContent = currentContent.clone();
+            }
+            currentContent.attr('id', currentTargetId + '_blockcontent');
 
-        var position = this.target.css("position");
-        if (position !== "fixed" && position !== "absolute") {
-            this.target.css('position', 'relative');
-        }
-        this.target.attr('aria-busy', this.cfg.blocked).append(this.blocker).append(this.content);
+            // assign data ids to this widget
+            currentBlocker.attr('data-bui-overlay', widgetId);
+            currentContent.attr('data-bui-content', widgetId);
+
+            // configure the target positioning
+            var position = currentTarget.css("position");
+            if (position !== "fixed" && position !== "absolute") {
+                currentTarget.css('position', 'relative');
+            }
 
-        if (this.target.length > 1) {
-            this.blocker = $(PrimeFaces.escapeClientId(this.id + '_blocker'));
-            this.content = this.target.children('.ui-blockui-content');
+            // ARIA 
+            currentTarget.attr('aria-busy', this.cfg.blocked);
+
+            // set the size and position to match the target
+            var height = currentTarget.height(),
+                width = currentTarget.width(),
+                position = currentTarget.position();
+            currentBlocker.css({
+                'height': height + 'px',
+                'width': width + 'px',
+                'left': position.left + 'px',
+                'top': position.top + 'px'
+            });
+            currentContent.css({
+                'height': height + 'px',
+                'width': width + 'px',
+                'left': position.left + 'px',
+                'top': position.top + 'px'
+            });
```

Comment:
The reason it was putting the block panel inside the target element was so it could properly calculated position so this just preemptively copies the size and position to the block UI elements.